### ISSUE TITLE
[13.0] Error: record.data[name].fieldsInfo is undefined

### ISSUE THREAD

**Issue by ieski** [odoo/odoo#65434]
Impacted versions:  13.0,

Steps to reproduce: Form does not open when BOM Component is clicked, error occurs

Current behavior: 

```
Traceback:
Error: record.data[name].fieldsInfo is undefined
_loadData/<@https://6013781-13-0.runbot36.odoo.com/web/content/1847-b0c5c41/web.assets_backend.js:1351:181
_.forEach@https://6013781-13-0.runbot36.odoo.com/web/content/1814-9240786/web.assets_common.js:108:566
_loadData@https://6013781-13-0.runbot36.odoo.com/web/content/1847-b0c5c41/web.assets_backend.js:1350:499

```

Expected behavior: Component Form needs to be opened

Video/Screenshot link (optional): https://pasteboard.co/JMxGhyL.jpg
![Screenshot 2021-02-02 235401](https://user-images.githubusercontent.com/88244/106661743-a1ea4100-65b2-11eb-8fde-08b3b9d3348e.jpeg)

Support ticket number submitted via odoo.com/help (optional):

**Comment by ieski** [comment]
**Bug line of code**
Source Code File:  addons\web\static\src\js\views\basic\basic_view.js 
Line Number: 141

**Comment by pedrobaeza** [comment]
I have gone to runbot, open BoM, and click on the component without problem, so unless you give more information on how to reproduce the problem, this is consider an invalid issue report.

**Comment by ieski** [comment]
> 
> 
> I have gone to runbot, open BoM, and click on the component without problem, so unless you give more information on how to reproduce the problem, this is consider an invalid issue report.

**Steps**
1- Manufacturing 
2- Master Data 
3- Bills Of Materials 
4- Expand group and click product from tree view 
5- Select Compound from the Form that opens and see the error

**Comment by pedrobaeza** [comment]
Can you please record a video, as I don't quite get the terminology you use

**Comment by ieski** [comment]
![Odoo](https://user-images.githubusercontent.com/88244/106712517-1ef4d500-660a-11eb-8eac-37ed496dbea7.gif)

https://user-images.githubusercontent.com/88244/106712523-20be9880-660a-11eb-84d1-005457a8b53f.mp4

It's simple, just follow the menu steps.

Thanks.

**Comment by valentinagiuliani1991** [comment]
Hi,
We have the same problem and we noticed that this happens if there is an invisible x2many field on the form (for example on the form view of purchase order line, the error occurs when from an order we try to open the detail form view of a line). We repeat this mistake on all of our customers and we need an urgent resolution

**Comment by pedrobaeza** [comment]
Tagged, but this doesn't guarantee that someone from Odoo is going to see it and solve it.

**Comment by vedler** [comment]
I've encountered this in development as well. One way to reproduce it is add a chatter to some object's form, then have some other model one2many reference it. When you try to open any of the o2m lines twice, the error breaks opening those record's form views until you refresh the page. I have my form view defined separately from the field itself (the form is not defined within the one2many, but the tree view is).

Chatter is just an easy example where there's an x2many hidden field on a popup form (target: new). Having two separate form views, one without the chatter defined in it (which is not shown in target-new views) is the fix we have used for now. I think I've only encountered it in Community version, but not Enterprise, but that might just be a coincidence.

When I checked it further, the _loadData function in [basic_view.js](https://github.com/odoo/odoo/blob/14.0/addons/web/static/src/js/views/basic/basic_view.js) is only called after the second click on the o2m line, but not the first. I checked an example from my code where there is a visible one2many in the popup form and the chatter (which is invisible), and it seems having the "mode" undefined on the x2mFieldInfo var for no-fetch (__no_fetch: True) one2many fields is what causes the issue, leaving the [viewType parameter](https://github.com/odoo/odoo/blob/14.0/addons/web/static/src/js/views/basic/basic_view.js#L149) inside the for-each loop in that function also undefined. That flag is set for invisible and column-invisible fields. 

![image](https://user-images.githubusercontent.com/22161082/107355704-87472900-6ad8-11eb-9e74-a6beb0733f8a.png)

Adding a x2mFieldInfo.__no_fetch check there *seems* to fix it, but I don't have a good understanding of the entire functionality there to say that it *is* the fix. My understanding is that the purpose of it is to reload data when those records on the inner one2many might have been modified elsewhere within the parent form, but I might be mistaken. Perhaps the one2many fetching should always define a viewType instead? Or maybe it would be correct to just skip any no-fetch fields there.
